### PR TITLE
Testing snippet

### PR DESCRIPTION
**Pull request by NimaSarajpoor**
This is to investigate `snippet` (tutorial notebook and the code) of the branch `main`

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -9,7 +9,9 @@ site_pkgs=$(python -c 'import site; print(site.getsitepackages()[0])')
 # Parse command line arguments
 for var in "$@"
 do
-    if [[ $var == "unit" ]]; then
+    if [[ $var == "check_notebooks" ]]; then
```
Comment (seanlaw) [inline test.sh:12]: Instead pf "check_notebooks" I would just call it "notebooks". So, from the command line it would be `./test.sh notebooks`

Comment (seanlaw) [reply test.sh:12]: I would add it after `coverage`

File: test.sh [hunk 2]
```diff
@@ -236,6 +238,15 @@ clean_up()
     fi
 }
 
+test_notebooks()
```
Comment (seanlaw) [inline test.sh:241]: The function name should be `convert_notebooks`

File: test.sh [hunk 3]
```diff
@@ -236,6 +238,15 @@ clean_up()
...
+{
+    echo "Testing notebooks"
+    for testfile in docs/Tutorial_*.ipynb
```
Comment (seanlaw) [inline test.sh:244]:
```
for notebook in `grep ipynb docs/tutorials.rst | sed -e 's/^[ \t]*//'`
do
    jupyter nbconvert --to notebook --execute "$notebook"
    check_errs $?
done```

File: test.sh [hunk 4]
```diff
@@ -236,6 +238,15 @@ clean_up()
...
+    do
+        jupyter nbconvert --to notebook --execute "$testfile"
+        check_errs $?
+    done
+}
 ###########
```
Comment (seanlaw) [inline test.sh:250]: Add a newline above

File: test.sh (hunk 3 above)
Comment (NimaSarajpoor) [reply test.sh:244]: So we are targetting the main tutorials. Cool.

**Pull request closed without merging**